native/common/jp_booleantype.cpp

```diff
@@ -218,3 +243,21 @@ void JPBooleanType::setArrayItem(JPJavaFrame& frame, jarray a, jsize ndx, PyObje
 	frame.SetBooleanArrayRegion(array, ndx, 1, &val);
 }
 
+void JPBooleanType::getView(JPArrayView& view)
+{
+	JPJavaFrame frame;
+	view.memory = (void*) frame.GetBooleanArrayElements(
+			(jbooleanArray) view.array->getJava(), &view.isCopy);
+	view.buffer.format = "?";
+	view.buffer.itemsize = sizeof (jboolean);
+}
+
+void JPBooleanType::releaseView(JPArrayView& view, bool complete)
+{
+	JPJavaFrame frame;
```

Comment:
you need the guard only if complete holds, right?

Reply:
The correct behavior is to do a release with commit (1) when write buffer holds and either an abort (2) or close (0) when the last view is written, but JNI_CHECK issues a segfault on the second commit (1).   I tried a number of experiments and honestly I don't get what JNI_CHECK.  The documentation in JNI specifically says COMMIT does not free the buffer but apparently that doesn't mean you can write to it some more and then release again as JNI has already washed its hands.     Perhaps I should just bite the bullet and refuse to share the buffers, but that created a really wacky side effect that if you get two memory views and wrote to the first and closed the first and the second the changes the first did get erased even if the second makes no changes.  

 CPython does not seem to request read only access when it should so the release is copying bytes when it shouldn't.  Hence a decision point, should we just always say we are read only and then file bug reports for users that request write access when they are only doing read ops.  (Or I am completely misreading the CPython API.)

Reply:
Does it segfault on OpenJVM? We could try to ask over there about implementation details. I honestly don't know. But I guess we have to commit every change once it is finished writing to actual buffer. 

The hassle with CPython has been dealt by the usage of NumPy, which wraps around some hassles of it. I'm again sorry, but I do not know the details here. Eventually you can look up some things in the C++ pybind11 project, since they have a well established NumPy layer (also dealing with memoryviews, I'd guess).

Reply:
Yes.  I am testing primarily on open JDK 11 using ubuntu on WLS and virtual image, with alternative testing on Oracle under windows.   Both show the same behavior.  I am guessing that my interpretation on "JNI_COMMIT" is incorrect.  It appears that it is just allowing you to keep the memory that was passed rather than allowing for a second commit.  But that seems like a very odd feature.

Because we can't commit more than once, we would get very odd behavior if we write back more than one behavior.  thus I recommend disabling the write until I can resolve this.  
I came up with several odd cases on write testing that leads me to think it would be a misfeature because you can defeat it simply by passing to another array that opens a buffer (such as asking for numpy to share an array.)  The safest course is to just issue readonly buffer as we never supported write through before.

If the user does actually want a write through they can use a direct buffer or the shared memory solution which are much more stablely supported and do not require pinning support.